enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/actions/SetRegionCommand.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.cloud.oracle.actions;
+
+import com.oracle.bmc.identity.Identity;
+import com.oracle.bmc.identity.IdentityClient;
+import com.oracle.bmc.identity.model.Region;
+import com.oracle.bmc.identity.requests.ListRegionsRequest;
+import com.oracle.bmc.identity.responses.ListRegionsResponse;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import org.netbeans.modules.cloud.oracle.OCIManager;
+import org.netbeans.modules.cloud.oracle.OCIProfile;
+import org.netbeans.spi.lsp.CommandProvider;
+import org.openide.DialogDisplayer;
+import org.openide.NotifyDescriptor;
+import org.openide.util.lookup.ServiceProvider;
+
+/**
+ * This command lets user select an Oracle Cloud region. This region will be used for all subsequent requests.
+ * 
+ * @author Jan Horvath
+ */
+@ServiceProvider(service = CommandProvider.class)
+public class SetRegionCommand implements CommandProvider {
+
+    private static final String SET_REGION_COMMAND = "nbls.oci.setRegion"; //NOI18N
+    private static final String GET_REGION_COMMAND = "nbls.oci.getRegion"; //NOI18N
+
+    private static Set<String> commands = new HashSet<String>(){{
+        add(SET_REGION_COMMAND);
+        add(GET_REGION_COMMAND);
+    }};
+
+    @Override
+    public Set<String> getCommands() {
+        return commands;
+    }
+
+    @Override
+    public CompletableFuture<Object> runCommand(String command, List<Object> arguments) {
+        CompletableFuture result = new CompletableFuture();
+        if (GET_REGION_COMMAND.equals(command)) {
+            result.complete(OCIManager.getDefault().getActiveSession().getRegion().getRegionCode());
+            return result;
+        }
+        if (SET_REGION_COMMAND.equals(command)) {
+            OCIProfile session = (OCIProfile) OCIManager.getDefault().getActiveSession();
+            Identity identityClient = session.newClient(IdentityClient.class);
+            ListRegionsRequest request = ListRegionsRequest.builder().build();
+
+            ListRegionsResponse response
```

Comment:
Looks like this call could block -- if so, pls. run in RequestProcessor complete the Future from within the offloaded task.